### PR TITLE
feat: allow adding more trusted principals to task role (users and ro…

### PR DESCRIPTION
**Pull request by kainlite**
## Description

This is adding another variable to be able to give some users or other roles the ability to assume the atlantis role, this can be refactored probably but I didn't want to add any breaking change for this simple requirement.

## Motivation and Context

While debugging it comes handy to be able to invoke terraform as atlantis does to speed things up, also this is particularly useful when importing resources.

## Breaking Changes

Optional variable, it should be a no op if empty.

## How Has This Been Tested?

Tested referencing my own branch.

### CODE REVIEW

**Comment by parmouraly** [comment]
Hey @kainlite 
I was just about to open an unrelated PR and came across this one here. 
Additional trusted principals are already supported
https://github.com/terraform-aws-modules/terraform-aws-atlantis/blob/master/variables.tf#L242-L246

**Comment by kainlite** [comment]
> Hey @kainlite
> 
> Additional trusted principals are already supported from release [2.27.0](https://github.com/terraform-aws-modules/terraform-aws-atlantis/blob/master/CHANGELOG.md#v2270---2020-12-10)
> 
> https://github.com/terraform-aws-modules/terraform-aws-atlantis/blob/master/variables.tf#L242-L246

Yes this is very similar, but the `trusted_principals` is only for AWS Services, my PR gives you the possibility to allow users or other roles to assume the atlantis role, this is particularly useful when importing, so your state get's the right permissions, owner, etc and you don't need to do any additional patching afterwards.

**Comment by parmouraly** [comment]
> Yes this is very similar, but the `trusted_principals` is only for AWS Services, my PR gives you the possibility to allow users or other roles to assume the atlantis role, this is particularly useful when importing, so your state gets the right permissions, owner, etc and you don't need to do any additional patching afterwards.

Got you, makes sense 👍 
I don't perform imports via Atlantis but I guess if you are then I can see the value.

**Comment by kainlite** [comment]
Updated from master

**Comment by OneBadSanta** [comment]
I need this functionality as well. This has been sitting for a while. Is there any technical reason why this has not been merged?

File: main.tf [hunk 1]
```diff
@@ -375,6 +375,11 @@ data "aws_iam_policy_document" "ecs_tasks" {
       type        = "Service"
       identifiers = compact(distinct(concat(["ecs-tasks.amazonaws.com"], var.trusted_principals)))
     }
+
+    principals {
+      type        = "AWS"
+      identifiers = compact(distinct(var.trusted_entities))
```
Comment (antonbabenko) [inline main.tf:381]: This code won't work if `trusted_entities` is empty. Please rewrite it to use `dynamic`.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (OneBadSanta) [reply main.tf:381]: That is indeed a good reason :) Thanks for the quick response. I will rewrite it

Comment (OneBadSanta) [reply main.tf:381]: @antonbabenko I wasn't sure how to contribute to this PR so i created another. Hope that's OK.

https://github.com/terraform-aws-modules/terraform-aws-atlantis/pull/193

**Comment by antonbabenko** [comment]
Please use [v2.37.0](https://github.com/terraform-aws-modules/terraform-aws-atlantis/releases/v2.37.0) which has been just released (#193).

**Pull request closed without merging**